Assets/Scripts/Entity/Player/Sanity/SanityPostProcessing.cs

```diff
@@ -0,0 +1,181 @@
+        }
+
+        private void OnDisable()
+        {
+            PlayerState.OnPlayerSanityUpdate -= (int sanity) => _sanity = sanity;
+        }
+
+        /// <summary>
+        /// Pulses all pulses. Masks them. Sets current values in post processing profile.
+        /// </summary>
+        private void Update()
+        {
+            PulseAll();
+            MaskAll();
+
+            // update values in effects
+            _colorGrading.saturation.value = _currentSaturation;
+            _colorGrading.contrast.value = _currentContrast;
+            _motionBlur.shutterAngle.value = _currentShutterAngle;
+            _vignette.intensity.value = _currentVignetteIntensity;
+            _chromaticAberration.intensity.value = _currentAberrationIntensity;
+        }
+        
+        private void ClampAll()
+        {
+            if (vignettePulseIntensity + _vignetteBaseIntensity > 1)
+                vignettePulseIntensity = 1 - _vignetteBaseIntensity;
+            if (aberrationPulseIntensity + _aberrationBaseIntensity > 1)
+                aberrationPulseIntensity = 1 - _aberrationBaseIntensity;
+            Mathf.Clamp(maxShutterAngle, 0, 360);
+            Mathf.Clamp(saturation, -100, 100);
+            Mathf.Clamp(contrast, -100, 100);
+        }
+
+        /// <summary>
+        /// Imitates a different pulse for vignette andd aberration.
+        /// </summary>
+        private void PulseAll()
+        {
+            _vignettePulse = Pulse(_vignettePulse, vignettePulseFrequency, vignettePulseIntensity, 0);
+            _aberrationPulse = Pulse(_aberrationPulse, aberrationPulseFrequency, aberrationPulseIntensity, 1);
+        }
+
+        /// <summary>
+        /// Masks all values depending on the sanity level and the <see cref="intensityCurve"/>.
+        /// </summary>
+        private void MaskAll()
+        {
+            float y = intensityCurve.Evaluate(_sanity / 100f);
+            _currentSaturation = (int)(y * saturation);
+            _currentContrast = (int)(y * contrast);
+            _currentShutterAngle = (int)(_baseShutterAngle + y * maxShutterAngle);
+            _currentVignetteIntensity = _vignetteBaseIntensity + y * _vignettePulse;
+            _currentAberrationIntensity = _aberrationBaseIntensity + y * _aberrationPulse;
+        }
+
+        /// <summary>
+        /// Pulse a value with a certain frequency and intensity.
+        /// </summary>
+        /// <param name="valueToBePulsed">The value pulsing should be applied to.</param>
+        /// <param name="pulseFrequency">The frequency of the pulse.</param>
+        /// <param name="pulseIntensity">The intensity of the pulse.</param>
+        /// <param name="boolIndex">Index of the boolean to be used for creating the pulse effect.</param>
+        /// <returns></returns>
+        private float Pulse(float valueToBePulsed, float pulseFrequency, float pulseIntensity, int boolIndex)
```

Comment:
I'd suggest using `ref bool isGrowing` for the last parameter, which basically works like a regular parameter and an out parameter combined. You can then call the method like this:
```c#
_vignettePulse = Pulse(_vignettePulse, vignettePulseFrequency, vignettePulseIntensity, _isGrowing[0]);
```
I think this makes the usage a little clearer and cleaner, since it doesn't depend on the array directly and you can avoid index out of bounds exceptions

Reply:
ahh that's what I was missing... ref